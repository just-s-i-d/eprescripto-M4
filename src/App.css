.pending {
  color: grey;
}

.cancelled {
  color: #f0675b;
  background-color: #feefee;
}

.done {
  color: #3fd6ab;
  background-color: #e6faf5;
}
