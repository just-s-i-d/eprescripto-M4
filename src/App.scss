.pending {
  color: $pendingColor;
}

.cancelled {
  color: $cancelledColor;
  background-color: $cancelledColorBg;
}

.done {
  color: $successColor;
  background-color: $successColorBg;
}
