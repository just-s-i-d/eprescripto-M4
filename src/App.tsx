import { useContext, useState } from "react";
import { Routes, Route } from "react-router-dom";
import { ConfigProvider, Spin } from "antd";

import PrivateRoute from "@routes/PrivateRoute";
import { UserContext } from "@context/UserProvider";
import DashboardLayout from "@layouts/DashboardLayout";
import RoleBasedRoutes from "@routes/RoleBasedRoutes";
import { darkTheme, lightTheme } from "./theme/theme.ts";
import "./App.scss";

function App() {
  const [isDarkTheme, setIsDarkTheme] = useState(false);
  const userContext = useContext(UserContext);
  if (userContext === null) return <Spin />;
  const { role } = userContext;
  return (
    <ConfigProvider theme={isDarkTheme ? darkTheme : lightTheme}>
      <Routes>
        <Route path="/" element={<PrivateRoute />}>
          <Route
            path="/"
            element={<DashboardLayout setIsDarkTheme={setIsDarkTheme} />}
          >
            <Route path="/*" element={<RoleBasedRoutes role={role} />} />
          </Route>
        </Route>
      </Routes>
    </ConfigProvider>
  );
}

export default App;
