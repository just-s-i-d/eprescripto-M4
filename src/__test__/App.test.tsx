import { render } from "@testing-library/react";
import App from "../App";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import "./matchMedia";

describe("App component tests", () => {
  it("Snapshot test", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
