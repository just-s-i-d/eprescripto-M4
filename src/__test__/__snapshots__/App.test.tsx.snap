// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="ant-layout w-full h-screen flex items-center justify-center css-dev-only-do-not-override-1yr7evd"
  >
    <div
      class="ant-card custom-card custom-card-login w-6/12 xs:h-[75vh] md:h-[65vh] rounded-[30px] z-20 xs:w-[95%] md:w-[91%] md:min-w-[700px] lg:w-[70%] xl:w-[65%] xxl:w-[55%] 2xl:h-[60vh] css-dev-only-do-not-override-1yr7evd"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="ant-space css-dev-only-do-not-override-1yr7evd ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small xs:w-full xs:mb-8 xs:h-[20%] md:mb-0 md:w-[45%] bg-primary rounded-[30px] px-6 md:h-full flex justify-center items-center"
        >
          <div
            class="ant-space-item"
          >
            <h1
              class="text-center text-4xl text-white mb-4 xs:mt-4 md:mt-0  md:block"
            >
              ePrescripto
            </h1>
          </div>
        </div>
        <div
          class="ant-space css-dev-only-do-not-override-1yr7evd ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small w-full h-5/6 px-8"
        >
          <div
            class="ant-space-item"
          >
            <div
              class="text-[34px]"
            >
              Sign In
            </div>
          </div>
          <div
            class="ant-space-item"
          >
            <form
              class="ant-form ant-form-vertical ant-form-large css-dev-only-do-not-override-1yr7evd custom-form flex flex-col gap-4 mt-6"
              data-testid="sign-in-form"
            >
              <div
                class="ant-form-item css-dev-only-do-not-override-1yr7evd"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1yr7evd"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1yr7evd"
                  >
                    <label
                      class="ant-form-item-required"
                      for="email"
                      title="Email"
                    >
                      Email
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1yr7evd"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-lg css-dev-only-do-not-override-1yr7evd"
                          id="email"
                          placeholder="ex. John@example.com"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item css-dev-only-do-not-override-1yr7evd"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1yr7evd"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1yr7evd"
                  >
                    <label
                      class="ant-form-item-required"
                      for="password"
                      title="Password"
                    >
                      Password
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1yr7evd"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <span
                          class="ant-input-affix-wrapper ant-input-password ant-input-affix-wrapper-lg ant-input-affix-wrapper-has-feedback css-dev-only-do-not-override-1yr7evd"
                        >
                          <input
                            aria-required="true"
                            class="ant-input ant-input-lg css-dev-only-do-not-override-1yr7evd"
                            id="password"
                            placeholder="password"
                            type="password"
                            value=""
                          />
                          <span
                            class="ant-input-suffix"
                          >
                            <span
                              aria-label="eye-invisible"
                              class="anticon anticon-eye-invisible ant-input-password-icon"
                              role="img"
                              tabindex="-1"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="eye-invisible"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                />
                                <path
                                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="ant-btn css-dev-only-do-not-override-1yr7evd ant-btn-default ant-btn-lg text-white w-[90px] mx-auto bg-primary disabled:opacity-50"
                disabled=""
                type="submit"
              >
                <span>
                  Sign In
                </span>
              </button>
            </form>
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="ant-divider css-dev-only-do-not-override-1yr7evd ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center w-9/12 min-w-9/12"
              role="separator"
            >
              <span
                class="ant-divider-inner-text"
              >
                Or
              </span>
            </div>
            <div
              class="flex justify-center"
            >
              <a
                class="ant-btn css-dev-only-do-not-override-1yr7evd ant-btn-default flex items-center gap-2 py-5 px-8 text-base bg-white"
                href="http://localhost:1337/api/connect/google"
              >
                <div
                  class="ant-image css-dev-only-do-not-override-1yr7evd"
                  style="width: 20px;"
                >
                  <img
                    alt="google"
                    class="ant-image-img css-dev-only-do-not-override-1yr7evd"
                    src="google.svg"
                    width="20"
                  />
                </div>
                <span>
                   Sign in with Google
                </span>
              </a>
            </div>
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="absolute md:bottom-8 xs:bottom-6"
            >
              Don't have an Account ?
              <span
                class="text-base cursor-pointer text-blue-500 ml-2"
              >
                Sign Up
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
